src/components/page/ProfilePage.tsx

```diff
@@ -1,10 +1,39 @@
 import { MobileLayout } from '@/components/layouts/MobileLayout';
 import { HeaderMenuCircleButton } from '@/components/parts/Header';
+import { UserProfile } from '@/features/profile/components';
+import { getCurrentUserId } from '@/lib/api/auth/server';
+import { getUserById } from '@/lib/api/user/server';
+
+export const ProfilePage = async () => {
+  /**
+   * Get the user ID of the logged-in user based on the auth token.
+   * Using the user ID, fetch the user's information from the server.
+   * @returns  The user's name on success. If the request fails,throws an error.
```

Comment:
@kanta1207 
The description is still technically incorrect because what throws an error on failure is not this function itself, but either `getCurrentUserId` or `getUserById`.

So, I don't think you need to specify the failure case for now.

Reply:
@nick-y-ito 
Thank you for sharing your point!
I suppose neither `getCurrentUserId` nor `getUserById` throws error, since these functions handle the error and return `result-ts-tyoe` "Err". 
So, I'm actually throwing an error declatively in line 29 of this file, inside of this function `getUserName` itself.
⬇️ 
```ts
  if (getCurrentUserIdResult.ok) {
      const { userId } = getCurrentUserIdResult.unwrap();
      const getUserByIdResult = await getUserById(userId);
      if (getUserByIdResult.ok) {
        const { name } = getUserByIdResult.unwrap();
        return name;
      }
    }
    throw new Error('Failed to get user name');
```

Reply:
Error handling in API client doesn't throw errors.
```ts
 } catch (error) {
    if (error instanceof Error) {
      return Err(error.message);
    }
    return Err('API response is invalid');
  }
```

Reply:
Oh, I'm sorry. I was probably watching the old code 😅

Looks good then! Thank you for fixing it!

Reply:
No worries. Thanks for not hesitating to share your thoughts as always!!